sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOPatientEverything.java

```diff
@@ -0,0 +1,317 @@
+      }
+    }
+
+    private Result(PCollectionTuple pct) {
+      this.pct = pct;
+      this.patientCompartments = pct.get(OUT).setCoder(JsonArrayCoder.of());
+      this.failedReads =
+          pct.get(DEAD_LETTER).setCoder(HealthcareIOErrorCoder.of(StringUtf8Coder.of()));
+    }
+
+    /**
+     * Gets failed reads.
+     *
+     * @return the failed reads
+     */
+    public PCollection<HealthcareIOError<String>> getFailedReads() {
+      return failedReads;
+    }
+
+    /**
+     * Gets the patient compartment responses for GetPatientEverything requests.
+     *
+     * @return the read patient compartments
+     */
+    public PCollection<JsonArray> getPatientCompartments() {
+      return patientCompartments;
+    }
+
+    @Override
+    public Pipeline getPipeline() {
+      return this.pct.getPipeline();
+    }
+
+    @Override
+    public Map<TupleTag<?>, PValue> expand() {
+      return ImmutableMap.of(OUT, patientCompartments, DEAD_LETTER, failedReads);
+    }
+
+    @Override
+    public void finishSpecifyingOutput(
+        String transformName, PInput input, PTransform<?, ?> transform) {}
+  }
+
+  @Override
+  public Result expand(PCollection<PatientEverythingParameter> input) {
+    PCollectionTuple results =
+        input.apply(
+            "GetPatientEverything",
+            ParDo.of(new GetPatientEverythingFn())
+                .withOutputTags(OUT, TupleTagList.of(DEAD_LETTER)));
+    return new Result(results);
+  }
+
+  /** GetPatientEverythingFn executes a GetPatientEverything request. */
+  static class GetPatientEverythingFn extends DoFn<PatientEverythingParameter, JsonArray> {
+
+    private static final Counter GET_PATIENT_EVERYTHING_ERROR_COUNT =
+        Metrics.counter(
+            GetPatientEverythingFn.class,
+            BASE_METRIC_PREFIX + "get_patient_everything_error_count");
+    private static final Counter GET_PATIENT_EVERYTHING_SUCCESS_COUNT =
+        Metrics.counter(
+            GetPatientEverythingFn.class,
+            BASE_METRIC_PREFIX + "get_patient_everything_success_count");
+    private static final Distribution GET_PATIENT_EVERYTHING_LATENCY_MS =
+        Metrics.distribution(
+            GetPatientEverythingFn.class, BASE_METRIC_PREFIX + "get_patient_everything_latency_ms");
+    private static final Logger LOG = LoggerFactory.getLogger(GetPatientEverythingFn.class);
+
+    @SuppressWarnings("initialization.fields.uninitialized")
+    private HealthcareApiClient client;
+
+    /**
+     * Instantiate healthcare client.
+     *
+     * @throws IOException the io exception
+     */
+    @Setup
+    public void instantiateHealthcareClient() throws IOException {
+      this.client = new HttpHealthcareApiClient();
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext context) {
+      PatientEverythingParameter patientEverythingParameter = context.element();
+      try {
+        context.output(
```

Comment:
if you want to parallelize the records of a single bundle, you could run something like that:

```
@StartBundle
tpe = new ThreadPoolExecutor();

@ProcessBundle
future = tpe.submit(param -> getPatientEverything(....))
futures.add(future)

@FinishBundle
for (f : futures) {
  // collect outputs or errors
}
```

for other healthcare IO we decided to not parallelize things - and this may be a fair choice in this case - but I'm just curious if you think this may be useful

Reply:
Will follow up for this.